server/services/project_chat_service.py

```diff
@@ -381,7 +385,46 @@ def _normalize_text(text: str) -> str:
 
 
 def _is_echo(user_input: str, candidate: str) -> bool:
```

Comment:
For each of the magic numbers definited in this function, extract them to named constants to improve code clarity